# Projet 3: [Réalisez une étude de santé publique](#M0) - Questions 1 à 14

# Table des matières
**[Question 1:](#M1)**
- [Calcul du nombre total d’humains sur la planète en 2013 et en 2017](#M1)

**[Question 2:](#M2)**  
- [1. Formule mathématique faisant intervenir chacune des 11 quantités données](#M21)
- [2. Illustration par le blé en France en 2013](#M22)

**[Question 3:](#M3)**
- [Calcul pour chaque pays et chaque produit de la disponibilité alimentaire en kcal et en kg de protéines](#M3)

**[Question 4:](#M4)**
- [1. Calcul pour chaque pays et chaque produit des ratios Energie/Poids](#M41)
- [2. Calcul pour chaque pays et chaque produit du pourcentage de protéines](#M42)

**[Question 5:](#M5)**
- [1. 5 aliments parmi les 20 aliments les plus caloriques, en utilisant le ratio énergie/poids](#M51)
- [2. 5 aliments parmi les 20 aliments les plus protéiniques, en utilisant le pourcentage de protéines](#M52)

**[Question 6:](#M6)**
- [Calcul pour les produits végétaux de la disponibilité intérieure mondiale exprimée en kcal](#M6)

**[Question 7:](#M7)**
- [Nombre d'humains nourris si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture:](#M7)
    - [1. a) en termes de calories](#M711)
    - [1. b) en pourcentage de la population mondiale](#M712)
    - [2. a) en termes de protéines](#M721)
    - [2. b) en pourcentage de la population mondiale](#M722)

**[Question 8:](#M8)**
- [Calcul du nombre d'humains nourris avec uniquement la nourriture de produits végétaux, leurs pertes et les aliments végétaux donnés aux animaux (exprimé en kcal et en kg de protéines)](#M8)

**[Question 9:](#M9)**
- [1. 5 aliments parmi les 20 aliments les plus caloriques, en utilisant le ratio énergie/poids](#M101)

**[Question 10:](#M10)**
- [1. 5 aliments parmi les 20 aliments les plus caloriques, en utilisant le ratio énergie/poids](#M102)

**[Question 11:](#M11)**
- [1. Proportion destinée à l'alimentation animale en terme de poids en 2013](#M111)
- [2. Sélection parmi les données des bilans alimentaires des informations relatives aux pays dans lesquels la FAO recense des personnes en sous-nutrition pour 2013.](#M112)
- [3. Les 15 produits les plus exportés par ce groupe de pays sur 2013](#M113)
- [4. Parmi les données des bilans alimentaires au niveau mondial, sélection des 200 plus grandes importations de ces produits en 2013.](#M114)
- [5. Groupez ces importations par produit, afin d'avoir une table contenant 1 ligne pour chacun des 15 produits.](#M115)
- [6. Calcul pour chaque produit des 2 quantités suivantes :](#M116)
    - [a) Calcul pour chaque produit des 2 quantités suivantes :](#M1161)
    - [b) le ratio entre la quantité destinée à la nourriture animale et la quantité destinée à la nourriture (animale + humaine)](#M1162)
    
**[Question 12:](#M12)**
- [3 produits ayant la plus grande valeur pour chacun des 2 ratios](#M12)

**[Question 13:](#M13)**
- [Tonnes de céréales libérées si les USA diminent leur production de produits animaux de 10%](#M13)

**[Question 14:](#M14)**
- [1. Proportion de manioc exportée par la Thaïlande en 2013](#M141)
- [2. Proportion de personnes en sous-nutrition en Thaïlande en 2013 et 2017](#M142)




In [1]:
import pandas as pd
import numpy as np

# Question 1:<a name="M1"></a>
## <span style='color:orange'>Calcul du nombre total d’humains sur la planète en 2013 et en 2017 <a name="M1"></a>

In [2]:
popu = pd.read_csv("population_entre_2012_et_2017.csv")
popu.head()

Code Domaine                       Domaine  Code zone         Zone  \
0           OA  Séries temporelles annuelles          2  Afghanistan   
1           OA  Séries temporelles annuelles          2  Afghanistan   
2           OA  Séries temporelles annuelles          2  Afghanistan   
3           OA  Séries temporelles annuelles          2  Afghanistan   
4           OA  Séries temporelles annuelles          2  Afghanistan   

   Code Élément            Élément  Code Produit                 Produit  \
0           511  Population totale          3010  Population-Estimations   
1           511  Population totale          3010  Population-Estimations   
2           511  Population totale          3010  Population-Estimations   
3           511  Population totale          3010  Population-Estimations   
4           511  Population totale          3010  Population-Estimations   

   Code année  Année           Unité     Valeur Symbole  \
0        2012   2012  1000 personnes  30696.958       X   
1        2013   2013  1000 personnes  31731.688       X   
2        2014   2014  1000 personnes  32758.020       X   
3        2015   2015  1000 personnes  33736.494       X   
4        2016   2016  1000 personnes  34656.032       X   

          Description du Symbole  \
0  Sources internationales sûres   
1  Sources internationales sûres   
2  Sources internationales sûres   
3  Sources internationales sûres   
4  Sources internationales sûres   

                                                Note  
0  UNDESA, Population Division – World Population...  
1  UNDESA, Population Division – World Population...  
2  UNDESA, Population Division – World Population...  
3  UNDESA, Population Division – World Population...  
4  UNDESA, Population Division – World Population...

In [3]:
popu = popu.drop(['Code Domaine','Domaine','Code zone','Code Élément','Élément','Code Produit','Produit','Code année','Unité','Symbole','Description du Symbole','Note'],1)
popu = popu.loc[popu['Année'].isin(["2013", "2017"])]
popu['Valeur'] *=1000
popu = pd.pivot_table(popu, values = "Valeur", index = ['Zone'], columns = ["Année"])
popu = popu.astype(int)
popu.head()

Année               2013      2017
Zone                              
Afghanistan     31731688  35530081
Afrique du Sud  53767395  56717155
Albanie          2918978   2930187
Algérie         38338562  41318142
Allemagne       81265139  82114224

In [4]:
popu = popu.drop_duplicates() # suppression des doublons (s'il en existe) 
np.sum(popu)

Année
2013    8627069068
2017    8991167322
dtype: int64

In [5]:
# Après observation, suppression des lignes qui faussent le calcul
popu = popu.drop(["Chine - RAS de Hong-Kong", "Chine - RAS de Macao", "Chine, Taiwan Province de", "Chine, continentale"])

In [6]:
# Calcul de la population mondiale en 2013 et en 2017
np.sum(popu)

Année
2013    7213211588
2017    7550036019
dtype: int64

<span style="color: blue">Résultats cohérents</span>


# Question 2:<a name="M2"></a>
## <span style='color:orange'>1. Formule mathématique faisant intervenir chacune des 11 quantités données<a name="M21"></a>

Disponibilité intérieure = Production + Importations - Exportations + Variation de stock = Nourritures + Traitements + Autres Utilisations + Pertes + Aliments pour Animaux + Semences

## <span style='color:orange'>2. Illustration par le blé en France en 2013<a name="M22"></a>

In [7]:
ble = pd.read_csv("cereales.csv")
ble.head()

Code Domaine              Domaine  Code Pays         Pays  Code Élément  \
0          FBS  Bilans Alimentaires          2  Afghanistan          5511   
1          FBS  Bilans Alimentaires          2  Afghanistan          5611   
2          FBS  Bilans Alimentaires          2  Afghanistan          5072   
3          FBS  Bilans Alimentaires          2  Afghanistan          5301   
4          FBS  Bilans Alimentaires          2  Afghanistan          5527   

                    Élément  Code Produit Produit  Code Année  Année  \
0                Production          2511     Blé        2013   2013   
1   Importations - Quantité          2511     Blé        2013   2013   
2        Variation de stock          2511     Blé        2013   2013   
3  Disponibilité intérieure          2511     Blé        2013   2013   
4                  Semences          2511     Blé        2013   2013   

                Unité  Valeur Symbole Description du Symbole  
0  Milliers de tonnes  5169.0       S  Données standardisées  
1  Milliers de tonnes  1173.0       S  Données standardisées  
2  Milliers de tonnes  -350.0       S  Données standardisées  
3  Milliers de tonnes  5992.0       S  Données standardisées  
4  Milliers de tonnes   322.0       S  Données standardisées

In [8]:
ble = ble.drop(['Code Domaine', 'Domaine', 'Code Pays', 'Code Élément', 'Code Produit', 'Code Année', 'Année', 'Symbole', 'Description du Symbole'],1)
ble = ble.loc[ble['Pays'].isin(["France"])].loc[ble['Produit'].isin(["Blé"])]
ble = pd.pivot_table(ble, 'Valeur', 'Élément', ['Pays', 'Produit'])
ble = ble.drop(["Disponibilité alimentaire (Kcal/personne/jour)", "Disponibilité alimentaire en quantité (kg/personne/an)", "Disponibilité de matière grasse en quantité (g/personne/jour)", "Disponibilité de protéines en quantité (g/personne/jour)"], 0).astype(int)
ble

Pays                     France
Produit                     Blé
Élément                        
Aliments pour animaux      7822
Autres Utilisations        2824
Disponibilité intérieure  20298
Exportations - Quantité   21502
Importations - Quantité    2055
Nourriture                 6971
Pertes                      358
Production                38614
Semences                    748
Traitement                 1575
Variation de stock         1131

In [9]:
ble.loc["Disponibilité intérieure"] == ble.loc["Production"] + ble.loc["Importations - Quantité"] - ble.loc["Exportations - Quantité"] + ble.loc["Variation de stock"]

Pays    Produit
France  Blé        True
dtype: bool

In [10]:
ble.loc["Disponibilité intérieure"] == ble.loc["Aliments pour animaux"] + ble.loc["Autres Utilisations"] + ble.loc["Nourriture"] + ble.loc["Pertes"] + ble.loc["Semences"] + ble.loc["Traitement"]

Pays    Produit
France  Blé        True
dtype: bool

<span style="color: blue">Egalités vérifiées!</span>

# Question 3:<a name="M3"></a>
## <span style='color:orange'>Calcul pour chaque pays et chaque produit de la disponibilité alimentaire en kcal et en kg de protéines<a name="M3"></a>

In [11]:
from fonctions_P3 import bil

In [12]:
dispos = bil(['Disponibilité de protéines en quantité (g/personne/jour)','Disponibilité alimentaire (Kcal/personne/jour)'])
dispos

Élément  \
Pays                                                              
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
Afghanistan      Disponibilité alimentaire (Kcal/personne/jour)   
Afghanistan   Disponibilité de protéines en quantité (g/pers...   
...                                                         ...   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon  Disponibilité de protéines en quantité (g/pers...   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon  Disponibilité de protéines en quantité (g/pers...   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon  Disponibilité de protéines en quantité (g/pers...   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon  Disponibilité de protéines en quantité (g/pers...   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon  Disponibilité de protéines en quantité (g/pers...   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon  Disponibilité de protéines en quantité (g/pers...   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon  Disponibilité de protéines en quantité (g/pers...   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon  Disponibilité de protéines en quantité (g/pers...   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon  Disponibilité de protéines en quantité (g/pers...   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon     Disponibilité alimentaire (Kcal/personne/jour)   
Îles Salomon  Disponibilité de proté

In [13]:
dispos['Valeur'] = dispos['Valeur'].mul(dispos['Population'], axis = 'rows')
dispos = pd.pivot_table(dispos, 'Valeur', ['Pays','Population','Produit'], 'Élément').fillna(0)
dispos["Disponibilité de protéines en quantité (g/personne/jour)"] /= 1000     # conversion des g en kg
dispos *= 365     # conversion des unités par jour en unités par année
dispos.columns = ["Disponibilité alimentaire (kcal)", "Disponibilité de protéines en quantité (kg)"]
dispos = dispos.apply(np.int64)
dispos.head(20)

Disponibilité alimentaire (kcal)  \
Pays        Population Produit                                                     
Afghanistan 31731688   Abats Comestible                              57910330600   
                       Agrumes, Autres                               11582066120   
                       Aliments pour enfants                         11582066120   
                       Ananas                                                  0   
                       Bananes                                       46328264480   
                       Beurre, Ghee                                 266387520760   
                       Bière                                                   0   
                       Blé                                        15855848518280   
                       Boissons Alcooliques                                    0   
                       Café                                                    0   
                       Coco (Incl Coprah)                                      0   
                       Crème                                         34746198360   
                       Céréales, Autres                                        0   
                       Dattes                                                  0   
                       Edulcorants Autres                            23164132240   
                       Feve de Cacao                                           0   
                       Fruits, Autres                               104238595080   
                       Graisses Animales Crue                       138984793440   
                       Huil Plantes Oleif Autr                       23164132240   
                       Huile Graines de Coton                        34746198360   

                                                Disponibilité de protéines en quantité (kg)  
Pays        Population Produit                                                               
Afghanistan 31731688   Abats Comestible                                             8918190  
                       Agrumes, Autres                                               231641  
                       Aliments pour enfants                                         347461  
                       Ananas                                                             0  
                       Bananes                                                       579103  
                       Beurre, Ghee                                                  347461  
                       Bière                                                              0  
                       Blé                                                        427494060  
                       Boissons Alcooliques                                               0  
                       Café                                                               0  
                       Coco (Incl Coprah)                                                 0  
                       Crème                                                         463282  
                       Céréales, Autres                                                   0  
                       Dattes                                                             0  
                       Edulcorants Autres                                                 0  
                       Feve de Cacao                                                 115820  
                       Fruits, Autres                                               1621489  
                       Graisses Animales Crue                                        115820  
                       Huil Plantes Oleif Autr                                            0  
                       Huile Graines de Coton                                             0

### <span style='color:orange'>Illustration avec Haïti:

In [14]:
dispos = dispos.reset_index()
dispos.loc[dispos['Pays'] == 'Haïti']

Pays  Population                   Produit  \
5145  Haïti    10431776          Abats Comestible   
5146  Haïti    10431776           Agrumes, Autres   
5147  Haïti    10431776     Aliments pour enfants   
5148  Haïti    10431776                    Ananas   
5149  Haïti    10431776  Animaux Aquatiques Autre   
5150  Haïti    10431776    Arachides Decortiquees   
5151  Haïti    10431776                    Avoine   
5152  Haïti    10431776                   Bananes   
5153  Haïti    10431776         Bananes plantains   
5154  Haïti    10431776              Beurre, Ghee   
5155  Haïti    10431776                     Bière   
5156  Haïti    10431776                       Blé   
5157  Haïti    10431776      Boissons Alcooliques   
5158  Haïti    10431776        Boissons Fermentés   
5159  Haïti    10431776                      Café   
5160  Haïti    10431776              Cephalopodes   
5161  Haïti    10431776           Citrons & Limes   
5162  Haïti    10431776        Coco (Incl Coprah)   
5163  Haïti    10431776                 Crustacés   
5164  Haïti    10431776                     Crème   
5165  Haïti    10431776          Céréales, Autres   
5166  Haïti    10431776        Edulcorants Autres   
5167  Haïti    10431776             Feve de Cacao   
5168  Haïti    10431776            Fruits, Autres   
5169  Haïti    10431776                  Girofles   
5170  Haïti    10431776    Graines Colza/Moutarde   
5171  Haïti    10431776    Graisses Animales Crue   
5172  Haïti    10431776                  Haricots   
5173  Haïti    10431776   Huil Plantes Oleif Autr   
5174  Haïti    10431776    Huile Graines de Coton   
...     ...         ...                       ...   
5197  Haïti    10431776            Patates douces   
5198  Haïti    10431776                 Perciform   
5199  Haïti    10431776        Plantes Aquatiques   
5200  Haïti    10431776  Plantes Oleiferes, Autre   
5201  Haïti    10431776                      Pois   
5202  Haïti    10431776        Poissons Eau Douce   
5203  Haïti    10431776   Poissons Marins, Autres   
5204  Haïti    10431776       Poissons Pelagiques   
5205  Haïti    10431776                    Poivre   
5206  Haïti    10431776                    Pommes   
5207  Haïti    10431776           Pommes de Terre   
5208  Haïti    10431776              Racines, Nda   
5209  Haïti    10431776                    Raisin   
5210  Haïti    10431776          Riz (Eq Blanchi)   
5211  Haïti    10431776                    Seigle   
5212  Haïti    10431776                      Soja   
5213  Haïti    10431776                    Sorgho   
5214  Haïti    10431776             Sucre Eq Brut   
5215  Haïti    10431776      Sucre non centrifugé   
5216  Haïti    10431776              Sucre, canne   
5217  Haïti    10431776                    Sésame   
5218  Haïti    10431776                       Thé   
5219  Haïti    10431776                   Tomates   
5220  Haïti    10431776    Viande d'Ovins/Caprins   
5221  Haïti    10431776          Viande de Bovins   
5222  Haïti    10431776          Viande de Suides   
5223  Haïti    10431776       Viande de Volailles   
5224  Haïti    10431776             Viande, Autre   
5225  Haïti    10431776                       Vin   
5226  Haïti    10431776            Épices, Autres   

      Disponibilité alimentaire (kcal)  \
5145                       15230392960   
5146                                 0   
5147                        3807598240   
5148                        3807598240   
5149                                 0   
5150                       83767161280   
5151                                 0   
5152                      133265938400   
5153                      114227947200   
5154                                 0   
5155                                 0   
5156                      536871351840   
5157                      270339475040   
5158                                 0   
5159                        7615196480   
5160                                 0   
5161  

# Question 4:<a name='M4'></a>
## <span style='color:orange'>1. Calcul pour chaque pays et chaque produit des ratios Energie/Poids<a name='M41'></a>

In [15]:
calo1 = bil(['Disponibilité alimentaire (Kcal/personne/jour)'])
calo1['Valeur'] = calo1['Valeur'].mul(calo1['Population'], axis = 'rows')
calo1 = pd.pivot_table(calo1, "Valeur", ['Pays','Population','Produit'], "Élément").fillna(0)
calo1.head()

Élément                                       Disponibilité alimentaire (Kcal/personne/jour)
Pays        Population Produit                                                              
Afghanistan 31731688   Abats Comestible                                          158658440.0
                       Agrumes, Autres                                            31731688.0
                       Aliments pour enfants                                      31731688.0
                       Ananas                                                            0.0
                       Bananes                                                   126926752.0

In [16]:
calo2 = bil(['Nourriture'])
calo2 = pd.pivot_table(calo2, "Valeur", ['Pays', 'Population', 'Produit'], "Élément").fillna(0)
calo = calo1.join(calo2)
calo["Disponibilité alimentaire (Kcal/personne/jour)"] *= 365
calo['Nourriture'] *= 1000000
calo.columns = ["Disponibilité alimentaire (kcal)", 'Nourriture (millions de kg)']
calo = calo.apply(np.int64)
calo = calo[(calo[['Disponibilité alimentaire (kcal)', 'Nourriture (millions de kg)']] > 0).all(axis = 1)] # supprime les zéros des colonnes (utile pour le futur calcul du ratio)
calo['Ratio "énergie/poids" (kcal/kg)'] = calo['Disponibilité alimentaire (kcal)'].div(calo['Nourriture (millions de kg)'], fill_value = 0)
calo = calo.reset_index()
calo = calo.drop(['Disponibilité alimentaire (kcal)', 'Nourriture (millions de kg)', 'Population'],1)
calo['Ratio "énergie/poids" (kcal/kg)'] = calo['Ratio "énergie/poids" (kcal/kg)'].round(decimals = 0)
calo

Pays                  Produit  Ratio "énergie/poids" (kcal/kg)
0      Afghanistan         Abats Comestible                           1093.0
1      Afghanistan          Agrumes, Autres                            297.0
2      Afghanistan    Aliments pour enfants                           5791.0
3      Afghanistan                  Bananes                            565.0
4      Afghanistan             Beurre, Ghee                           7400.0
5      Afghanistan                      Blé                           3239.0
6      Afghanistan                    Crème                           1930.0
7      Afghanistan       Edulcorants Autres                           1448.0
8      Afghanistan           Fruits, Autres                            467.0
9      Afghanistan   Graisses Animales Crue                           9266.0
10     Afghanistan  Huil Plantes Oleif Autr                          11582.0
11     Afghanistan   Huile Graines de Coton                           8687.0
12     Afghanistan  Huile de Colza&Moutarde                          11582.0
13     Afghanistan           Huile de Palme                           9229.0
14     Afghanistan            Huile de Soja                           9410.0
15     Afghanistan          Huile de Sésame                          11582.0
16     Afghanistan       Huile de Tournesol                           9064.0
17     Afghanistan       Lait - Excl Beurre                            609.0
18     Afghanistan          Légumes, Autres                            222.0
19     Afghanistan      Légumineuses Autres                           3600.0
20     Afghanistan                     Maïs                           3200.0
21     Afghanistan                     Miel                           5791.0
22     Afghanistan                   Millet                           2896.0
23     Afghanistan                     Noix                           3159.0
24     Afghanistan                    Oeufs                           1448.0
25     Afghanistan      Oranges, Mandarines                            320.0
26     Afghanistan                     Orge                           3384.0
27     Afghanistan                   Pommes                            511.0
28     Afghanistan          Pommes de Terre                            755.0
29     Afghanistan                   Raisin                            695.0
...            ...                      ...                              ...
8754      Éthiopie                     Soja                           3298.0
8755      Éthiopie                   Sorgho                           3116.0
8756      Éthiopie            Sucre Eq Brut                           3572.0
8757      Éthiopie                   Sésame                           6927.0
8758      Éthiopie   Viande d'Ovins/Caprins                           2106.0
8759      Éthiopie         Viande de Bovins                           2241.0
8760      Éthiopie      Viande de Volailles                           1117.0
8761      Éthiopie            Viande, Autre                           1226.0
8762      Éthiopie           Épices, Autres                           3848.0
8763  Îles Salomon                    Bière                            411.0
8764  Îles Salomon                      Blé                           2703.0
8765  Îles Salomon                     Café                            206.0
8766  Îles Salomon       Coco (Incl Coprah)                           1116.0
8767  Îles Salomon           Fruits, Autres                            433.0
8768  Îles Salomon           Huile de Palme                          10695.0
8769  Îles Salomon                  Ignames                            935.0
8770  Îles Salomon       Lait - Excl Beurre                            651.0
8771  Îles Salomon          Légumes, Autres                            235.0
8772  Îles Salomon      Légumineuses Autres                           3394.0
8773  Îles Salomon                   Manioc                           1028.0
8774  Îles

In [17]:
calo.loc[calo['Produit'].isin(['Bière'])].head()

Pays Produit  Ratio "énergie/poids" (kcal/kg)
46   Afrique du Sud   Bière                            501.0
102         Albanie   Bière                            446.0
151         Algérie   Bière                            447.0
208       Allemagne   Bière                            481.0
277          Angola   Bière                            524.0

In [18]:
calo_oeuf = calo.loc[calo['Produit'].isin(['Oeufs'])]
calo_oeuf.head()

Pays Produit  Ratio "énergie/poids" (kcal/kg)
24      Afghanistan   Oeufs                           1448.0
74   Afrique du Sud   Oeufs                           1446.0
124         Albanie   Oeufs                           1225.0
177         Algérie   Oeufs                           1196.0
240       Allemagne   Oeufs                           1380.0

In [19]:
calo_oeuf.describe().apply(np.int64)

Ratio "énergie/poids" (kcal/kg)
count                              166
mean                              1355
std                                171
min                                683
25%                               1282
50%                               1377
75%                               1444
max                               2019

In [20]:
calo_oeuf.loc[calo['Pays'] == 'France']

Pays Produit  Ratio "énergie/poids" (kcal/kg)
2660  France   Oeufs                           1410.0

## <span style='color:orange'>2) Calcul pour chaque pays et chaque produit du pourcentage de protéines<a name='M42'></a>

In [21]:
prot = bil(['Disponibilité de protéines en quantité (g/personne/jour)', 'Disponibilité alimentaire en quantité (kg/personne/an)'])
prot = prot.drop(['Population'], 1)
prot = pd.pivot_table(prot, 'Valeur', ['Pays', 'Produit'], 'Élément').fillna(0)
prot["Disponibilité de protéines en quantité (g/personne/jour)"] *= 365
prot["Disponibilité alimentaire en quantité (kg/personne/an)"] *= 1000
prot.columns = ["Disponibilité alimentaire en quantité (g/personne/an)","Disponibilité de protéines en quantité (g/personne/an)"]
prot = prot[(prot[['Disponibilité alimentaire en quantité (g/personne/an)', 'Disponibilité de protéines en quantité (g/personne/an)']] != 0).all(axis = 1)]  # supprime les zéros des colonnes (utile pour le futur calcul du ratio)
prot['Pourcentage de protéines'] = (prot['Disponibilité de protéines en quantité (g/personne/an)'].mul(100)).div(prot['Disponibilité alimentaire en quantité (g/personne/an)'], fill_value = 0)
prot = prot.reset_index()
prot = prot.drop(['Disponibilité alimentaire en quantité (g/personne/an)','Disponibilité de protéines en quantité (g/personne/an)'],1)
prot.head().round(decimals = 2)

Pays                Produit  Pourcentage de protéines
0  Afghanistan       Abats Comestible                     16.34
1  Afghanistan        Agrumes, Autres                      0.57
2  Afghanistan  Aliments pour enfants                     18.25
3  Afghanistan                Bananes                      0.68
4  Afghanistan           Beurre, Ghee                      0.94

In [22]:
prot.loc[prot['Produit'] == 'Avoine'].describe().round(decimals = 2)

Pourcentage de protéines
count                    114.00
mean                       8.76
std                        1.64
min                        5.09
25%                        8.11
50%                        8.70
75%                        9.12
max                       18.25

In [23]:
prot.loc[prot['Produit'] == 'Oeufs'].describe().round(decimals = 2)

Pourcentage de protéines
count                    171.00
mean                      10.69
std                        0.53
min                        9.73
25%                       10.37
50%                       10.69
75%                       11.22
max                       12.08

In [24]:
prot.loc[prot['Produit'] == 'Viande de Bovins'].describe().round(decimals = 2)

Pourcentage de protéines
count                    171.00
mean                      14.46
std                        1.37
min                        9.40
25%                       13.89
50%                       14.61
75%                       14.90
max                       21.17

In [25]:
prot.loc[prot['Produit'] == 'Épices, Autres'].describe().round(decimals = 2)

Pourcentage de protéines
count                    155.00
mean                      11.09
std                        2.39
min                        5.98
25%                        9.65
50%                       10.87
75%                       12.17
max                       21.90

In [26]:
prot.loc[prot['Produit'] == 'Sésame'].describe().round(decimals = 2)

Pourcentage de protéines
count                     98.00
mean                      18.74
std                        4.06
min                       12.17
25%                       17.59
50%                       18.25
75%                       18.25
max                       36.50

In [27]:
prot.loc[prot['Produit'] == 'Riz (Eq Blanchi)'].describe().round(decimals = 2)

Pourcentage de protéines
count                    171.00
mean                       7.01
std                        0.51
min                        5.74
25%                        6.71
50%                        6.92
75%                        7.20
max                        8.80

# Question 5:<a name='M5'></a>
## <span style='color:orange'>1. 5 aliments parmi les 20 aliments les plus caloriques, en utilisant le ratio énergie/poids<a name='M51'></a>

In [28]:
rang_cal = pd.pivot_table(calo, 'Ratio "énergie/poids" (kcal/kg)', 'Pays', 'Produit')
rang_cal.head()

Produit         Abats Comestible  Agrumes, Autres  Aliments pour enfants  \
Pays                                                                       
Afghanistan               1093.0            297.0                 5791.0   
Afrique du Sud            1313.0              NaN                    NaN   
Albanie                   1065.0              NaN                    NaN   
Algérie                    982.0              NaN                 3816.0   
Allemagne                  957.0            158.0                    NaN   

Produit         Ananas  Animaux Aquatiques Autre  Arachides Decortiquees  \
Pays                                                                       
Afghanistan        NaN                       NaN                     NaN   
Afrique du Sud   456.0                       NaN                  5681.0   
Albanie            NaN                       NaN                     NaN   
Algérie            NaN                       NaN                  6361.0   
Allemagne        543.0                       NaN                  5650.0   

Produit         Avoine  Bananes  Bananes plantains  Beurre, Ghee  ...  Sésame  \
Pays                                                              ...           
Afghanistan        NaN    565.0                NaN        7400.0  ...  6515.0   
Afrique du Sud  2453.0    635.0              685.0        7230.0  ...     NaN   
Albanie            NaN    533.0                NaN        6126.0  ...     NaN   
Algérie            NaN    567.0                NaN        7696.0  ...  5597.0   
Allemagne       3036.0    605.0                NaN        7016.0  ...     NaN   

Produit           Thé  Tomates  Viande d'Ovins/Caprins  Viande de Bovins  \
Pays                                                                       
Afghanistan     433.0      NaN                  2394.0            2234.0   
Afrique du Sud    NaN    194.0                  2171.0            2288.0   
Albanie           NaN    161.0                  1848.0            1516.0   
Algérie           NaN    209.0                  1999.0            1590.0   
Allemagne       503.0    207.0                  2472.0            1062.0   

Produit         Viande de Suides  Viande de Volailles  Viande, Autre    Vin  \
Pays                                                                          
Afghanistan                  NaN               1390.0          965.0    NaN   
Afrique du Sud            3747.0               1494.0         1280.0  675.0   
Albanie                   2527.0               1700.0            NaN  639.0   
Algérie                      NaN               1250.0         1000.0    NaN   
Allemagne                 1640.0               1312.0         1013.0  673.0   

Produit         Épices, Autres  
Pays                            
Afghanistan             2896.0  
Afrique du Sud          2804.0  
Albanie                    NaN  
Algérie                 2799.0  
Allemagne               3490.0  

[5 rows x 94 columns]

In [29]:
# on crée un dataframe avec la moyenne de chaque produit pour tous les pays
rang_cal = pd.DataFrame(rang_cal.mean().apply(np.int64))
rang_cal.head()

0
Produit                       
Abats Comestible          1140
Agrumes, Autres            253
Aliments pour enfants     3714
Ananas                     445
Animaux Aquatiques Autre   456

In [30]:
# on retient les 20 premiers rangs en les triant dans l'ordre décroissant
rang_cal.sort_values(by = 0, ascending = False).iloc[0:20]

0
Produit                        
Huiles de Foie de Poisso  10000
Huile de Sésame            9406
Huile de Son de Riz        9381
Huile d'Arachide           9036
Huiles de Poissons         9024
Huile de Germe de Maïs     9001
Huil Plantes Oleif Autr    8977
Huile d'Olive              8966
Huile de Palmistes         8923
Huile Graines de Coton     8874
Huile de Palme             8812
Huile de Tournesol         8756
Huile de Colza&Moutarde    8755
Huile de Soja              8734
Huile de Coco              8575
Graisses Animales Crue     7672
Beurre, Ghee               7407
Palmistes                  6553
Sésame                     5884
Arachides Decortiquees     5751

<span style="color: blue">On peut donc citer par exemple l'huie de foie de poisson, l'huile d'olive, l'huile de tournesol, le sésame et les arachides décortiquées.

## <span style='color: orange'>2. 5 aliments parmi les 20 aliments les plus protéiniques, en utilisant le pourcentage de protéines<a name='M52'></a>

In [31]:
rang_prot = pd.pivot_table(prot, 'Pourcentage de protéines', 'Pays', 'Produit')
rang_prot.head()

Produit         Abats Comestible  Agrumes, Autres  Aliments pour enfants  \
Pays                                                                       
Afghanistan            16.340116         0.565891              18.250000   
Afrique du Sud         19.577963              NaN                    NaN   
Albanie                16.685714              NaN              16.222222   
Algérie                15.858621              NaN              15.263636   
Allemagne              18.006667         0.160793                    NaN   

Produit           Ananas  Animaux Aquatiques Autre  Arachides Decortiquees  \
Pays                                                                         
Afghanistan          NaN                       NaN                     NaN   
Afrique du Sud  0.450617                       NaN               24.676056   
Albanie              NaN                       NaN               26.766667   
Algérie              NaN                       NaN               27.375000   
Allemagne       0.330816                       NaN               25.866142   

Produit            Avoine   Bananes  Bananes plantains  Beurre, Ghee  ...  \
Pays                                                                  ...   
Afghanistan           NaN  0.675926                NaN      0.935897  ...   
Afrique du Sud   7.786667  0.985085           0.901235      1.042857  ...   
Albanie          8.423077  0.701923                NaN      0.920168  ...   
Algérie               NaN  0.984921                NaN      0.730000  ...   
Allemagne       11.376623  0.800844                NaN      0.919574  ...   

Produit            Sésame        Thé   Tomates  Viande d'Ovins/Caprins  \
Pays                                                                     
Afghanistan     18.951923  10.040115       NaN               14.169715   
Afrique du Sud        NaN  10.428571  1.027633               13.047897   
Albanie               NaN   4.562500  0.831820               13.484987   
Algérie         18.250000   9.954545  1.053911               13.337451   
Allemagne             NaN   7.197183  0.979221               12.500000   

Produit         Viande de Bovins  Viande de Suides  Viande de Volailles  \
Pays                                                                      
Afghanistan            15.029412               NaN            12.961224   
Afrique du Sud         14.636439          9.739183            14.427069   
Albanie                15.184000          9.158548            13.628874   
Algérie                14.665179               NaN            11.215633   
Allemagne              13.063450         10.046130            14.929014   

Produit         Viande, Autre  Vin  Épices, Autres  
Pays                                                
Afghanistan         16.328947  NaN       15.442308  
Afrique du Sud      18.250000  NaN       14.038462  
Albanie             18.250000  NaN        9.125000  
Algérie             15.208333  NaN       13.140000  
Allemagne           20.604839  NaN       11.573171  

[5 rows x 88 columns]

In [32]:
rang_prot = pd.DataFrame(rang_prot.mean().round(decimals = 1))
rang_prot.head()

0
Produit                       
Abats Comestible          17.6
Agrumes, Autres            0.4
Aliments pour enfants     15.4
Ananas                     0.3
Animaux Aquatiques Autre   4.6

In [33]:
rang_prot.sort_values(by = 0, ascending = False).iloc[0:20]

0
Produit                       
Graines Colza/Moutarde    36.5
Soja                      27.2
Arachides Decortiquees    25.1
Pois                      22.4
Légumineuses Autres       22.3
Haricots                  21.9
Plantes Oleiferes, Autre  18.8
Sésame                    18.7
Viande, Autre             17.8
Abats Comestible          17.6
Aliments pour enfants     15.4
Cephalopodes              15.1
Viande de Bovins          14.5
Viande d'Ovins/Caprins    13.9
Viande de Volailles       12.6
Piments                   12.4
Poissons Pelagiques       12.3
Graines de tournesol      12.0
Poissons Marins, Autres   11.3
Poivre                    11.2

<span style="color: blue">On peut citer le soja, les arachides décortiquées, les haricots, le sésame et les piments. </span>

# Question 6:<a name='M6'></a>
## <span style='color: orange'>Calcul pour les produits végétaux de la disponibilité intérieure mondiale exprimée en kcal<a name='M6'></a>

In [34]:
from fonctions_P3 import Veget

In [35]:
dispo_veget = Veget(['Disponibilité intérieure'])
dispo_veget = pd.pivot_table(dispo_veget, 'Valeur', ['Pays', 'Produit'], 'Élément')
dispo_veget['Disponibilité intérieure'] *= 1000000    # conversion des milliers de tonnes en kg
dispo_veget.columns = ['Disponibilité intérieure en kg']
dispo_veget.head()

Disponibilité intérieure en kg
Pays        Produit                                              
Afghanistan Agrumes, Autres                            41000000.0
            Aliments pour enfants                       2000000.0
            Ananas                                            0.0
            Bananes                                    82000000.0
            Bière                                       3000000.0

In [36]:
ratio = pd.pivot_table(calo, values = 'Ratio "énergie/poids" (kcal/kg)', index = ['Pays','Produit'])
ratio.head()

Ratio "énergie/poids" (kcal/kg)
Pays        Produit                                               
Afghanistan Abats Comestible                                1093.0
            Agrumes, Autres                                  297.0
            Aliments pour enfants                           5791.0
            Bananes                                          565.0
            Beurre, Ghee                                    7400.0

In [37]:
dispo_int_kcal = dispo_veget.join(ratio).fillna(0)
dispo_int_kcal['Disponibilité intérieur en kcal'] = dispo_int_kcal['Disponibilité intérieure en kg'].mul(dispo_int_kcal['Ratio "énergie/poids" (kcal/kg)'], fill_value = 0)
dispo_int_kcal = dispo_int_kcal[(dispo_int_kcal[['Disponibilité intérieur en kcal']] != 0).all(axis = 1)]
dispo_int_kcal = dispo_int_kcal.drop(['Disponibilité intérieure en kg','Ratio "énergie/poids" (kcal/kg)'],1)
dispo_int_kcal.head()

Disponibilité intérieur en kcal
Pays        Produit                                               
Afghanistan Agrumes, Autres                           1.217700e+10
            Aliments pour enfants                     1.158200e+10
            Bananes                                   4.633000e+10
            Blé                                       1.940809e+13
            Edulcorants Autres                        2.316800e+10

In [38]:
np.sum(dispo_int_kcal)

Disponibilité intérieur en kcal    1.236036e+16
dtype: float64

<span style="color: blue">Soit plus de 12 millions de milliards de kcal de disponibilité intérieure mondiale de produits végétaux pour 2013</span>

 # Question 7:<a name='M7'></a>
## <span style='color: orange'>Nombre d'humains nourris si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture:<a name='M7'></a>

### <span style='color: green'>1. a) en termes de calories<a name='M711'></a>

Recommandation concernant le nombre de calories par jour nécessaire à un être humain: 2586 kcal/personne/jour<br/> (Normes fondées sur les rapports du Comité spécial mixte FAO/OMS/UNU d'experts sur les besoins en énergie et en protéines)

In [39]:
val_moy_kcal = 2586

La disponibilité intérieure mondiale de 2013 est de 12 360 750 000 000 000 kcal.

In [40]:
popu_dispo_int_kcal = np.sum(dispo_int_kcal)/val_moy_kcal/365
popu_dispo_int_kcal

Disponibilité intérieur en kcal    1.309512e+10
dtype: float64

<span style="color: blue">Soit quasiment 13,1 milliards d'humains nourris uniquement avec la disponibilité intérieure mondiale des produits végétaux!</span>

### <span style='color: green'>1. b) en pourcentage de la population mondiale<a name='M712'></a>

Population en 2013: 7 213 211 588 humains

In [41]:
popu_dispo_int_kcal/np.sum(popu).iloc[0]*100

Disponibilité intérieur en kcal    181.543586
dtype: float64

<span style="color: blue">Soit environ 182 % de la population mondiale en 2013, i.e. presque deux fois la population mondiale.</span>

### <span style='color: green'>2. a) en terme de protéines<a name='M721'></a>

In [42]:
dispo_veget = Veget(['Disponibilité intérieure'])
dispo_veget = pd.pivot_table(dispo_veget, 'Valeur', ['Pays', 'Produit'], 'Élément')
dispo_veget['Disponibilité intérieure'] *= 1000000000 # conversion de milliers de tonnes en grammes
dispo_veget.columns = ['Disponibilité intérieure en g']
dispo_veget.head()

Disponibilité intérieure en g
Pays        Produit                                             
Afghanistan Agrumes, Autres                         4.100000e+10
            Aliments pour enfants                   2.000000e+09
            Ananas                                  0.000000e+00
            Bananes                                 8.200000e+10
            Bière                                   3.000000e+09

In [43]:
rang_prot = pd.pivot_table(prot, values = 'Pourcentage de protéines', index = ['Pays','Produit'])
rang_prot.head()

Pourcentage de protéines
Pays        Produit                                        
Afghanistan Abats Comestible                      16.340116
            Agrumes, Autres                        0.565891
            Aliments pour enfants                 18.250000
            Bananes                                0.675926
            Beurre, Ghee                           0.935897

In [44]:
dispo_int_prot = dispo_veget.join(rang_prot).fillna(0)
dispo_int_prot['Disponibilité intérieure de protéines en g'] = (dispo_int_prot['Disponibilité intérieure en g'].div(100)).mul(dispo_int_prot['Pourcentage de protéines'], fill_value = 0)
dispo_int_prot = dispo_int_prot.drop(['Disponibilité intérieure en g','Pourcentage de protéines'],1)
dispo_int_prot.head()

Disponibilité intérieure de protéines en g
Pays        Produit                                                          
Afghanistan Agrumes, Autres                                      2.320155e+08
            Aliments pour enfants                                3.650000e+08
            Ananas                                               0.000000e+00
            Bananes                                              5.542593e+08
            Bière                                                0.000000e+00

In [45]:
np.sum(dispo_int_prot)

Disponibilité intérieure de protéines en g    2.917255e+14
dtype: float64

Le taux de protéines recommandé est de 0,83 g/kg/jour pour un humain moyen (selon la FA0).<br/>
Soit une moyenne de 49,2 g/jour/personne pour un poids moyen de 59,25 kg.

In [46]:
val_moy_prot = 49.2

In [47]:
popu_dispo_int_prot = np.sum(dispo_int_prot)/val_moy_prot/365
popu_dispo_int_prot

Disponibilité intérieure de protéines en g    1.624488e+10
dtype: float64

<span style="color: blue">Soit quasiment 16,2 milliards d'humains nourris uniquement avec la disponibilité intérieure mondiale de protéines pour les produits végétaux.</span>

### <span style='color: green'>2. b) en pourcentage de la population mondiale<a name='M722'></a>

In [48]:
popu_dispo_int_prot/np.sum(popu).iloc[0]*100

Disponibilité intérieure de protéines en g    225.210066
dtype: float64

<span style="color: blue">Soit 225 % de la population mondiale, i.e. plus du double de la population.</span>

# Question 8:<a name='M8'></a>
## <span style='color: orange'>Calcul du nombre d'humains nourris avec uniquement la nourriture de produits végétaux, leurs pertes et les aliments végétaux donnés aux animaux (exprimé en kcal et en kg de protéines)<a name='M8'></a>

In [49]:
veget = Veget(['Nourriture','Aliments pour animaux', 'Pertes'])
veget = pd.pivot_table(veget, 'Valeur', ['Pays', 'Produit'], 'Élément')
veget['Nourriture'] *= 1000000
veget['Aliments pour animaux'] *= 1000000
veget['Pertes'] *= 1000000

In [50]:
ratio = pd.pivot_table(calo, values = 'Ratio "énergie/poids" (kcal/kg)', index = ['Pays','Produit'])

In [51]:
proteines = pd.pivot_table(prot, values = 'Pourcentage de protéines', index = ['Pays','Produit'])

In [52]:
prod_agri = veget.join(ratio).join(proteines).fillna(0)
prod_agri['Disponibilité alimentaire en kcal'] = ((prod_agri['Aliments pour animaux'].add(prod_agri['Nourriture'])).add(prod_agri['Pertes'])).mul(prod_agri['Ratio "énergie/poids" (kcal/kg)'], fill_value = 0)
prod_agri['Disponibilité alimentaire de protéines en kg'] = (((prod_agri['Aliments pour animaux'].add(prod_agri['Nourriture'])).add(prod_agri['Pertes'])).div(100)).mul(prod_agri['Pourcentage de protéines'], fill_value = 0)
prod_agri = prod_agri.drop(['Aliments pour animaux', 'Nourriture', 'Pertes', 'Ratio "énergie/poids" (kcal/kg)', 'Pourcentage de protéines'],1)
prod_agri.head()

Disponibilité alimentaire en kcal  \
Pays        Produit                                                    
Afghanistan Agrumes, Autres                             1.217700e+10   
            Aliments pour enfants                       1.158200e+10   
            Ananas                                      0.000000e+00   
            Bananes                                     4.633000e+10   
            Bière                                       0.000000e+00   

                                   Disponibilité alimentaire de protéines en kg  
Pays        Produit                                                              
Afghanistan Agrumes, Autres                                       232015.503876  
            Aliments pour enfants                                 365000.000000  
            Ananas                                                     0.000000  
            Bananes                                               554259.259259  
            Bière                                                      0.000000

In [53]:
np.sum(prod_agri)

Disponibilité alimentaire en kcal               9.219981e+15
Disponibilité alimentaire de protéines en kg    2.042652e+11
dtype: float64

### <span style='color: green'>Avec la disponibilté alimentaire en kcal:<a name='M81'></a>

In [54]:
popu_prod_agri_kcal = np.sum(prod_agri).iloc[0]/val_moy_kcal/365
popu_prod_agri_kcal

9768066887.031328

<span style="color: blue">Soit presque 9,77 milliards d'humains nourris avec uniquement la nourriture de produits végétaux, leurs pertes et les aliments végétaux donnés aux animaux</span>

In [55]:
popu_prod_agri_kcal/np.sum(popu).iloc[0]*100

135.41910933656268

<span style="color: blue">Soit environ 135% de la population mondiale.</span>

### <span style='color: green'>Avec la disponibilité alimentaire de protéines en kg:<a name='M82'></a>

In [56]:
popu_prod_agri_prot = np.sum(prod_agri).iloc[1]/(val_moy_prot/1000)/365
popu_prod_agri_prot

11374607030.842045

<span style="color: blue">Soit presque 11,4 milliards de la population mondiale nourris avec uniquement la nourriture de produits végétaux, leurs pertes et les aliments végétaux donnés aux animaux</span>

In [57]:
popu_prod_agri_prot/np.sum(popu).iloc[0]*100

157.69129869647801

<span style="color: blue">Soit quasiment 158% de la population mondiale.</span>

# Question 9:<a name='M9'></a>
## <span style='color: orange'>Calcul du nombre d'humains nourris avec uniquement la disponibilité alimentaire mondiale (exprimé en kcal et en kg de protéines)<a name='M9'></a>

In [58]:
dispos_jour = bil(["Disponibilité de protéines en quantité (g/personne/jour)","Disponibilité alimentaire (Kcal/personne/jour)"])
dispos_jour['Valeur'] = dispos_jour['Valeur'].mul(dispos_jour['Population'], axis = 'rows')
dispos_jour = pd.pivot_table(dispos_jour, 'Valeur', ['Pays', 'Population', 'Produit'], 'Élément').fillna(0)
dispos_jour.columns = ["Disponibilité alimentaire (kcal/jour)", "Disponibilité de protéines en quantité (g/jour)"]
dispos_jour.head()

Disponibilité alimentaire (kcal/jour)  \
Pays        Population Produit                                                        
Afghanistan 31731688   Abats Comestible                                 158658440.0   
                       Agrumes, Autres                                   31731688.0   
                       Aliments pour enfants                             31731688.0   
                       Ananas                                                   0.0   
                       Bananes                                          126926752.0   

                                              Disponibilité de protéines en quantité (g/jour)  
Pays        Population Produit                                                                 
Afghanistan 31731688   Abats Comestible                                           24433399.76  
                       Agrumes, Autres                                              634633.76  
                       Aliments pour enfants                                        951950.64  
                       Ananas                                                            0.00  
                       Bananes                                                     1586584.40

In [59]:
np.sum(dispos_jour)

Disponibilité alimentaire (kcal/jour)              2.030097e+13
Disponibilité de protéines en quantité (g/jour)    5.708862e+11
dtype: float64

### <span style='color: green'>Avec la disponibilité alimentaire en kcal/jour:<a name='M91'></a>

In [60]:
popu_dispos_jour = np.sum(dispos_jour).iloc[0]/val_moy_kcal
popu_dispos_jour

7850336678.136891

<span style="color: blue">Soit environ 7,85 milliards d'humains nourris avec uniquement la disponibilité alimentaire mondiale.</span>

In [61]:
popu_dispos_jour/np.sum(popu).iloc[0]*100

108.83275199076125

<span style="color: blue">Soit un peu plus que la population mondiale.</span>

### <span style='color: green'>Avec la disponibilité de protéines en quantité (g/jour):<a name='M92'></a>

In [62]:
popu_dispos_jour = np.sum(dispos_jour).iloc[1]/val_moy_prot
popu_dispos_jour

11603378867.896576

<span style="color: blue">Soit environ 11,6 milliards d'humains nourris avec uniquement la disponibilité alimentaire mondiale.</span>

In [63]:
popu_dispos_jour/np.sum(popu).iloc[0]*100

160.86286567830783

<span style="color: blue">Soit environ 161% de la population mondiale</span>

# Question 10:<a name='M10'></a>
## <span style='color: orange'>1. Proportion de la population mondiale considérée comme étant en sous-nutrition en 2013 et en 2017<a name='M101'></a>

In [64]:
sous_alim = pd.read_csv('sous_alimentes.csv').fillna(0)
sous_alim = sous_alim.loc[sous_alim['Année'].isin(["2012-2014", "2016-2018"])]
sous_alim = pd.pivot_table(sous_alim, values = 'Valeur', index = ['Zone'], columns = ['Année'], aggfunc = sum)
sous_alim['2012-2014'] = pd.to_numeric(sous_alim['2012-2014'], errors='coerce') # pour convertir des '<0.1' en Nan
sous_alim['2016-2018'] = pd.to_numeric(sous_alim['2016-2018'], errors= 'coerce')
sous_alim = sous_alim.fillna(0.1) # ordre de grandeur de base avant conversion
sous_alim *= 1000000
sous_alim.columns = ['Nombre de sous-alimentés en 2013', 'Nombre de sous-alimentés en 2017']
sous_alim = sous_alim.drop(["Chine - RAS de Hong-Kong", "Chine - RAS de Macao", "Chine, Taiwan Province de", "Chine, continentale"])
sous_alim.head()

Nombre de sous-alimentés en 2013  \
Zone                                               
Afghanistan                            7900000.0   
Afrique du Sud                         2600000.0   
Albanie                                 200000.0   
Algérie                                1700000.0   
Allemagne                                    0.0   

                Nombre de sous-alimentés en 2017  
Zone                                              
Afghanistan                           10600000.0  
Afrique du Sud                         3500000.0  
Albanie                                 200000.0  
Algérie                                1600000.0  
Allemagne                                    0.0

In [65]:
np.sum(sous_alim)

Nombre de sous-alimentés en 2013    746000000.0
Nombre de sous-alimentés en 2017    753300000.0
dtype: float64

In [66]:
np.sum(sous_alim).iloc[0]/np.sum(popu).iloc[0]*100

10.342133887227932

<span style="color: blue">Soit 10,34 % de la population considérée comme étant en sous-nutrition en 2013</span>

In [67]:
np.sum(sous_alim).iloc[1]/np.sum(popu).iloc[1]*100

9.977435844071303

<span style="color: blue">Soit 9,98 % de la population considérée comme étant en sous-nutrition en 2017</span>

## <span style='color: orange'>2. Liste des produits, ainsi que leurs codes, considérés comme des céréales selon la FAO<a name='M102'></a>

In [68]:
cereales = pd.read_csv('cereales.csv')
cereales = cereales.drop(['Code Domaine', 'Domaine', 'Code Élément', 'Élément', 'Code Pays', 'Pays', 'Code Année', 'Année', 'Unité', 'Valeur', 'Symbole', 'Description du Symbole'],1)
cereales = cereales.groupby(['Code Produit', 'Produit']).agg(np.sum)
cereales

Empty DataFrame
Columns: []
Index: [(2511, Blé), (2513, Orge), (2514, Maïs), (2515, Seigle), (2516, Avoine), (2517, Millet), (2518, Sorgho), (2520, Céréales, Autres), (2805, Riz (Eq Blanchi))]

## <span style='color: orange'>3. Informations concernant les céréales en créant une colonne de type booléen nommée "is_cereal"<a name='M103'></a>

In [69]:
cereales = cereales.reset_index()
list(cereales['Code Produit'])

[2511, 2513, 2514, 2515, 2516, 2517, 2518, 2520, 2805]

In [74]:
veg = pd.read_csv("veg.csv")
veg = veg.drop(['Code Domaine', 'Domaine', 'Code Pays', 'Code Élément', 'Code Année', 'Symbole', 'Description du Symbole'],1)
veg = veg.loc[veg['Année'] == 2013 ]
veg = veg.drop(['Année'],1)
veg['is_cereal'] = veg.apply(lambda x: x['Code Produit'] in list(cereales['Code Produit']), axis = 1)
veg.insert(4, 'is_cereal', veg.pop('is_cereal'))    # pour placer la colone "is_cereal" derrière "Produit"
veg

Pays                                            Élément  \
1       Afghanistan                                         Production   
3       Afghanistan                            Importations - Quantité   
5       Afghanistan                                 Variation de stock   
7       Afghanistan                           Disponibilité intérieure   
9       Afghanistan                                           Semences   
11      Afghanistan                                             Pertes   
13      Afghanistan                                         Nourriture   
15      Afghanistan  Disponibilité alimentaire en quantité (kg/pers...   
17      Afghanistan     Disponibilité alimentaire (Kcal/personne/jour)   
19      Afghanistan  Disponibilité de protéines en quantité (g/pers...   
21      Afghanistan  Disponibilité de matière grasse en quantité (g...   
23      Afghanistan                                         Production   
25      Afghanistan                            Importations - Quantité   
27      Afghanistan                           Disponibilité intérieure   
29      Afghanistan                                           Semences   
31      Afghanistan                                             Pertes   
33      Afghanistan                                         Nourriture   
35      Afghanistan  Disponibilité alimentaire en quantité (kg/pers...   
37      Afghanistan     Disponibilité alimentaire (Kcal/personne/jour)   
39      Afghanistan  Disponibilité de protéines en quantité (g/pers...   
41      Afghanistan  Disponibilité de matière grasse en quantité (g...   
43      Afghanistan                                         Production   
45      Afghanistan                            Importations - Quantité   
47      Afghanistan                                 Variation de stock   
49      Afghanistan                           Disponibilité intérieure   
51      Afghanistan                              Aliments pour animaux   
53      Afghanistan                                           Semences   
55      Afghanistan                                             Pertes   
57      Afghanistan                                         Nourriture   
59      Afghanistan  Disponibilité alimentaire en quantité (kg/pers...   
...             ...                                                ...   
209683        Chine  Disponibilité alimentaire en quantité (kg/pers...   
209685        Chine     Disponibilité alimentaire (Kcal/personne/jour)   
209687        Chine  Disponibilité de protéines en quantité (g/pers...   
209689        Chine                                         Production   
209691        Chine                            Importations - Quantité   
209693        Chine                                 Variation de stock   
209695        Chine                            Exportations - Quantité   
209697        Chine                           Disponibilité intérieure   
209699        Chine                                Autres Utilisations   
209701        Chine                                         Nourriture   
209703        Chine  Disponibilité alimentaire en quantité (kg/pers...   
209705        Chine     Disponibilité alimentaire (Kcal/personne/jour)   
209707        Chine                                         Production   
209709        Chine                            Importations - Quantité   
209711        Chine                            Exportations - Quantité   
209713        Chine                           Disponibilité intérieure   
209715        Chine                                Autres Utilisations   
209717        Chine                            Importations - Quantité   
209719        Chine                                 Variation de stock   
209721        Chine                            Exportations - Quantité   
209723        Chine                           Disponibilité intérieure   
209725        Chine                                             Pertes   
209727        Chine

# Question 11:<a name='M11'></a>
## <span style='color: orange'>1. Proportion destinée à l'alimentation animale en terme de poids en 2013<a name='M111'></a>

In [75]:
from fonctions_P3 import Cereals

In [76]:
alim_anim = Cereals()
alim_anim = pd.pivot_table(alim_anim, 'Valeur', ['Pays', 'Produit'], 'Élément')
alim_anim = alim_anim.fillna(0)
alim_anim.head()

Élément                       Aliments pour animaux  Nourriture
Pays        Produit                                            
Afghanistan Blé                                 0.0      4895.0
            Céréales, Autres                    0.0         0.0
            Maïs                              200.0        76.0
            Millet                              0.0        12.0
            Orge                              360.0        89.0

In [77]:
np.sum(alim_anim)

Élément
Aliments pour animaux    1070617.0
Nourriture               1241405.0
dtype: float64

In [78]:
np.sum(alim_anim).iloc[0]/(np.sum(alim_anim).iloc[0]+np.sum(alim_anim).iloc[1])*100

46.30652303481541

<span style="color: blue">Soit 46,3% des céréales sont destinés à l'alimentation animale en 2013.</span>

## <span style='color: orange'>2. Sélection parmi les données des bilans alimentaires des informations relatives aux pays dans lesquels la FAO recense des personnes en sous-nutrition pour 2013.<a name='M112'></a>

In [79]:
from fonctions_P3 import Select

In [80]:
export = Select()
export = export[(export[['Sous-alimentés']] != 0).all(axis = 1)]
pd.pivot_table(export, values = 'Valeur', index = ['Pays', 'Sous-alimentés', 'Produit'], columns = ['Élément']).fillna(0).head()

Élément                                           Aliments pour animaux  \
Pays        Sous-alimentés Produit                                        
Afghanistan 7900000.0      Abats Comestible                         0.0   
                           Agrumes, Autres                          0.0   
                           Aliments pour enfants                    0.0   
                           Ananas                                   0.0   
                           Bananes                                  0.0   

Élément                                           Autres Utilisations  \
Pays        Sous-alimentés Produit                                      
Afghanistan 7900000.0      Abats Comestible                       0.0   
                           Agrumes, Autres                        0.0   
                           Aliments pour enfants                  0.0   
                           Ananas                                 0.0   
                           Bananes                                0.0   

Élément                                           Disponibilité alimentaire (Kcal/personne/jour)  \
Pays        Sous-alimentés Produit                                                                 
Afghanistan 7900000.0      Abats Comestible                                                  5.0   
                           Agrumes, Autres                                                   1.0   
                           Aliments pour enfants                                             1.0   
                           Ananas                                                            0.0   
                           Bananes                                                           4.0   

Élément                                           Disponibilité alimentaire en quantité (kg/personne/an)  \
Pays        Sous-alimentés Produit                                                                         
Afghanistan 7900000.0      Abats Comestible                                                    1.72        
                           Agrumes, Autres                                                     1.29        
                           Aliments pour enfants                                               0.06        
                           Ananas                                                              0.00        
                           Bananes                                                             2.70        

Élément                                           Disponibilité de matière grasse en quantité (g/personne/jour)  \
Pays        Sous-alimentés Produit                                                                                
Afghanistan 7900000.0      Abats Comestible                                                    0.20               
                           Agrumes, Autres                                                     0.01               
                           Aliments pour enfants                                               0.01               
                           Ananas                                                              0.00               
                           Bananes                                                             0.02               

Élément                                           Disponibilité de protéines en quantité (g/personne/jour)  \
Pays        Sous-alimentés Produit                                                                           
Afghanistan 7900000.0      Abats Comestible                                                    0.77          
                           Agrumes, Autres                                                     0.02          
                           Aliments pour enfants                                               0.03          
                           Ananas                                                              0.00          
                           Bananes    

## <span style='color: orange'>3. Les 15 produits les plus exportés par ce groupe de pays sur 2013<a name='M113'></a>

In [77]:
export = export.loc[export['Élément'].isin(['Exportations - Quantité'])]
export = export[(export[['Valeur']] != 0).all(axis = 1)]
export = export.drop(['Sous-alimentés'],1)
export = export.groupby(['Produit']).agg(np.sum)
export = export.sort_values(by = 'Valeur', ascending = False).iloc[0:15]
export

Valeur
Produit                     
Huile de Palme       46336.0
Maïs                 37906.0
Manioc               35931.0
Riz (Eq Blanchi)     33104.0
Sucre Eq Brut        26160.0
Blé                  25417.0
Légumes, Autres      20167.0
Bananes              18005.0
Fruits, Autres       15389.0
Lait - Excl Beurre   15060.0
Soja                 14410.0
Poissons Pelagiques  14185.0
Tomates               8938.0
Pommes                7958.0
Oranges, Mandarines   7185.0

## <span style='color: orange'>4. Parmi les données des bilans alimentaires au niveau mondial, sélection des 200 plus grandes importations de ces produits en 2013.<a name='M114'></a>

In [78]:
export = export.reset_index()
list(export['Produit'])

['Huile de Palme',
 'Maïs',
 'Manioc',
 'Riz (Eq Blanchi)',
 'Sucre Eq Brut',
 'Blé',
 'Légumes, Autres',
 'Bananes',
 'Fruits, Autres',
 'Lait - Excl Beurre',
 'Soja',
 'Poissons Pelagiques',
 'Tomates',
 'Pommes',
 'Oranges, Mandarines']

In [79]:
imports = Select()
imports = imports.drop(['Sous-alimentés'],1)
imports = imports.loc[imports['Élément'].isin(['Importations - Quantité'])]
imports = imports.loc[imports['Produit'].isin(list(export['Produit']))]
imports = imports.sort_values(by = 'Valeur', ascending = False).iloc[0:200]
imports

Élément             Produit               Unité  \
Pays                                                                        
Chine     Importations - Quantité                Soja  Milliers de tonnes   
Chine     Importations - Quantité              Manioc  Milliers de tonnes   
Japon     Importations - Quantité                Maïs  Milliers de tonnes   
Égypte    Importations - Quantité                 Blé  Milliers de tonnes   
Chine     Importations - Quantité  Lait - Excl Beurre  Milliers de tonnes   
...                           ...                 ...                 ...   
Pologne   Importations - Quantité                 Blé  Milliers de tonnes   
Canada    Importations - Quantité                Maïs  Milliers de tonnes   
Brésil    Importations - Quantité  Lait - Excl Beurre  Milliers de tonnes   
Autriche  Importations - Quantité                 Blé  Milliers de tonnes   
Chili     Importations - Quantité                 Blé  Milliers de tonnes   

           Valeur  
Pays               
Chine     65564.0  
Chine     30466.0  
Japon     14403.0  
Égypte    10331.0  
Chine      9780.0  
...           ...  
Pologne    1037.0  
Canada     1035.0  
Brésil     1032.0  
Autriche   1031.0  
Chili      1022.0  

[200 rows x 4 columns]

## <span style='color: orange'>5. Groupez ces importations par produit, afin d'avoir une table contenant 1 ligne pour chacun des 15 produits.<a name='M115'></a>

In [80]:
imports = pd.pivot_table(imports, 'Valeur', 'Produit', 'Pays').fillna(0)
imports

Pays                 Afghanistan  Afrique du Sud  Algérie  Allemagne  \
Produit                                                                
Bananes                      0.0             0.0      0.0     1344.0   
Blé                       1173.0          1490.0   6343.0     5421.0   
Fruits, Autres               0.0             0.0      0.0     2499.0   
Huile de Palme               0.0             0.0      0.0     2456.0   
Lait - Excl Beurre           0.0             0.0   2581.0     8332.0   
Légumes, Autres              0.0             0.0      0.0     4378.0   
Manioc                       0.0             0.0      0.0        0.0   
Maïs                         0.0             0.0   3251.0     2465.0   
Oranges, Mandarines          0.0             0.0      0.0     2147.0   
Poissons Pelagiques          0.0             0.0      0.0     1246.0   
Pommes                       0.0             0.0      0.0     1177.0   
Riz (Eq Blanchi)             0.0          1311.0      0.0        0.0   
Soja                         0.0             0.0      0.0     3620.0   
Sucre Eq Brut                0.0             0.0   1836.0        0.0   
Tomates                      0.0             0.0      0.0     1764.0   

Pays                 Arabie saoudite  Autriche  Azerbaïdjan  Bangladesh  \
Produit                                                                   
Bananes                          0.0       0.0          0.0         0.0   
Blé                           2484.0    1031.0       1589.0      2882.0   
Fruits, Autres                   0.0       0.0          0.0         0.0   
Huile de Palme                   0.0       0.0          0.0      1278.0   
Lait - Excl Beurre            2134.0    1123.0          0.0         0.0   
Légumes, Autres                  0.0       0.0          0.0         0.0   
Manioc                           0.0       0.0          0.0         0.0   
Maïs                          2120.0       0.0          0.0         0.0   
Oranges, Mandarines              0.0       0.0          0.0         0.0   
Poissons Pelagiques              0.0       0.0          0.0         0.0   
Pommes                           0.0       0.0          0.0         0.0   
Riz (Eq Blanchi)              1222.0       0.0          0.0         0.0   
Soja                             0.0       0.0          0.0         0.0   
Sucre Eq Brut                 1780.0       0.0          0.0      1727.0   
Tomates                          0.0       0.0          0.0         0.0   

Pays                 Belgique  Brésil  ...  Sénégal  Thaïlande  Tunisie  \
Produit                                ...                                
Bananes                1275.0     0.0  ...      0.0        0.0      0.0   
Blé                    4570.0  7630.0  ...      0.0     2118.0   1489.0   
Fruits, Autres            0.0     0.0  ...      0.0        0.0      0.0   
Huile de Palme            0.0     0.0  ...      0.0        0.0      0.0   
Lait - Excl Beurre     4332.0  1032.0  ...      0.0     1241.0      0.0   
Légumes, Autres        1603.0     0.0  ...      0.0        0.0      0.0   
Manioc                    0.0     0.0  ...      0.0     1250.0      0.0   
Maïs                   1606.0     0.0  ...      0.0        0.0      0.0   
Oranges, Mandarines    3615.0     0.0  ...      0.0        0.0      0.0   
Poissons Pelagiques       0.0     0.0  ...      0.0     1460.0      0.0   
Pommes                    0.0     0.0  ...      0.0        0.0      0.0   
Riz (Eq Blanchi)          0.0     0.0  ...   1120.0        0.0      0.0   
Soja                      0.0     0.0  ...      0.0     1682.0      0.0   
Sucre Eq Brut             0.0     0.0  ...      0.0        0.0      0.0   
Tomates                   0.0     0.0  ...      0.0        0.0      0.0   

Pays                 Turquie  Venezuela (République bolivarienne du)  \
Produit                                                                
Bananes                  0.0                                     0.0   
Blé    

## <span style='color: orange'>6. Calcul pour chaque produit des 2 quantités suivantes :<a name='M116'></a>

###  <span style='color: green'>a) le ratio entre la quantité destinés aux "Autres utilisations" et la disponibilité intérieure.<a name='M1161'></a>

In [81]:
imports = imports.reset_index()
list(imports['Produit'])

['Bananes',
 'Blé',
 'Fruits, Autres',
 'Huile de Palme',
 'Lait - Excl Beurre',
 'Légumes, Autres',
 'Manioc',
 'Maïs',
 'Oranges, Mandarines',
 'Poissons Pelagiques',
 'Pommes',
 'Riz (Eq Blanchi)',
 'Soja',
 'Sucre Eq Brut',
 'Tomates']

In [82]:
ratio1 = bil(['Autres Utilisations', 'Disponibilité intérieure'])
ratio1 = ratio1.drop(['Population'], 1)
ratio1 = ratio1.loc[ratio1['Produit'].isin(list(imports['Produit']))]
ratio1 = pd.pivot_table(ratio1, 'Valeur', ['Pays','Produit'], 'Élément').fillna(0)
ratio1['Ratio'] = (ratio1['Autres Utilisations'].mul(100)).div(ratio1['Disponibilité intérieure']).fillna(0).round(decimals = 2)
ratio1

Élément                          Autres Utilisations  \
Pays         Produit                                   
Afghanistan  Bananes                             0.0   
             Blé                                 0.0   
             Fruits, Autres                      0.0   
             Huile de Palme                     55.0   
             Lait - Excl Beurre                  0.0   
...                                              ...   
Îles Salomon Pommes                              0.0   
             Riz (Eq Blanchi)                   12.0   
             Soja                                0.0   
             Sucre Eq Brut                       0.0   
             Tomates                             0.0   

Élément                          Disponibilité intérieure  Ratio  
Pays         Produit                                              
Afghanistan  Bananes                                 82.0   0.00  
             Blé                                   5992.0   0.00  
             Fruits, Autres                         240.0   0.00  
             Huile de Palme                         119.0  46.22  
             Lait - Excl Beurre                    2083.0   0.00  
...                                                   ...    ...  
Îles Salomon Pommes                                   0.0   0.00  
             Riz (Eq Blanchi)                        49.0  24.49  
             Soja                                     0.0   0.00  
             Sucre Eq Brut                            2.0   0.00  
             Tomates                                  0.0   0.00  

[2537 rows x 3 columns]

### <span style='color: green'>b) le ratio entre la quantité destinée à la nourriture animale et la quantité destinée à la nourriture (animale + humaine)<a name='M1162'></a>

In [83]:
ratio2 = bil(['Aliments pour animaux', 'Nourriture'])
ratio2 = ratio2.drop(['Population'], 1)
ratio2 = ratio2.loc[ratio2['Produit'].isin(list(imports['Produit']))]
ratio2 = pd.pivot_table(ratio2, 'Valeur', ['Pays','Produit'], 'Élément').fillna(0)
ratio2['Ratio'] = (ratio2['Aliments pour animaux'].mul(100)).div(ratio2['Nourriture'].add(ratio2['Aliments pour animaux'])).fillna(0).round(decimals = 2)
ratio2.head()

Élément                         Aliments pour animaux  Nourriture  Ratio
Pays        Produit                                                     
Afghanistan Bananes                               0.0        82.0   0.00
            Blé                                   0.0      4895.0   0.00
            Fruits, Autres                        0.0       223.0   0.00
            Huile de Palme                        0.0        64.0   0.00
            Lait - Excl Beurre                  123.0      1901.0   6.08

# Question 12:<a name='M12'></a>
## <span style='color: orange'>3 produits ayant la plus grande valeur pour chacun des 2 ratios<a name='M12'></a>

In [84]:
ratio1.groupby(['Produit']).max().sort_values(by = 'Ratio', ascending = False).iloc[0:3]

Élément              Autres Utilisations  Disponibilité intérieure  Ratio
Produit                                                                  
Huile de Palme                    6782.0                    8413.0  200.0
Manioc                            9257.0                   53004.0  100.0
Poissons Pelagiques                141.0                   10161.0  100.0

In [85]:
ratio2.groupby(['Produit']).max().sort_values(by = 'Ratio', ascending = False).iloc[0:3]

Élément              Aliments pour animaux  Nourriture  Ratio
Produit                                                      
Manioc                             26288.0     20636.0  100.0
Maïs                              153802.0     14232.0  100.0
Poissons Pelagiques                 9575.0      2896.0  100.0

# Question 13:<a name='M13'></a>
## <span style='color: orange'>Tonnes de céréales libérées si les USA diminent leur production de produits animaux de 10%<a name='M13'></a>

In [86]:
cereal_ani = Cereals()
cereal_ani = cereal_ani.loc[cereal_ani['Pays'] == "États-Unis d'Amérique"].loc[cereal_ani['Élément'].isin(['Aliments pour animaux'])]
cereal_ani = pd.pivot_table(cereal_ani, 'Valeur', ['Pays', 'Élément', 'Produit'])
cereal_ani

Valeur
Pays                  Élément               Produit                   
États-Unis d'Amérique Aliments pour animaux Avoine              1430.0
                                            Blé                 6196.0
                                            Céréales, Autres      77.0
                                            Maïs              128024.0
                                            Millet               394.0
                                            Orge                1440.0
                                            Seigle               184.0
                                            Sorgho              2351.0

In [87]:
np.sum(cereal_ani)*1000/10 # conversion de milliers de tonnes en tonnes et calcul du pourcentage (10%)

Valeur    14009600.0
dtype: float64

<span style="color: blue">Soit environ 14 millions de tonnes de céréales libérées pour 2013 aux USA</span>

# Question 14:<a name='M14'></a>
## <span style='color: orange'>1. Proportion de manioc exportée par la Thaïlande en 2013<a name='M141'></a>

In [88]:
manioc_prod = Select().reset_index().drop(['Sous-alimentés'],1)
manioc_prod = manioc_prod.loc[manioc_prod['Pays'] =='Thaïlande'].loc[manioc_prod['Élément'].isin(['Production'])].loc[manioc_prod['Produit'] == 'Manioc']
manioc_prod

Pays     Élément Produit               Unité   Valeur
121908  Thaïlande  Production  Manioc  Milliers de tonnes  30228.0

In [89]:
manioc_export = Select().reset_index().drop(['Sous-alimentés'],1)
manioc_export = manioc_export.loc[manioc_export['Pays'] =='Thaïlande'].loc[manioc_export['Élément'].isin(['Exportations - Quantité'])].loc[manioc_export['Produit'] == 'Manioc']
manioc_export

Pays                  Élément Produit               Unité  \
121911  Thaïlande  Exportations - Quantité  Manioc  Milliers de tonnes   

         Valeur  
121911  25214.0

In [90]:
manioc_export.loc[121911,'Valeur']/manioc_prod.loc[121908,'Valeur']*100

83.41272991928014

<span style="color: blue">Le manioc représente environ 83% de l'exportation du pays</span>

## <span style='color: orange'>2. Proportion de personnes en sous-nutrition en Thaïlande en 2013 et 2017<a name='M142'></a>

In [91]:
popu = pd.read_csv("population_entre_2012_et_2017.csv")
popu = popu.drop(['Domaine','Élément','Produit','Note'],1).loc[popu['Année'].isin(["2013", "2017"])]
popu['Valeur'] *=1000
popu_thai = popu.loc[popu['Zone'] == 'Thaïlande']
popu_thai

Code Domaine  Code zone       Zone  Code Élément  Code Produit  \
1279           OA        216  Thaïlande           511          3010   
1283           OA        216  Thaïlande           511          3010   

      Code année  Année           Unité      Valeur Symbole  \
1279        2013   2013  1000 personnes  68143065.0       X   
1283        2017   2017  1000 personnes  69037513.0       X   

             Description du Symbole  
1279  Sources internationales sûres  
1283  Sources internationales sûres

In [92]:
sousal = pd.read_csv('sous_alimentes.csv').fillna(0)
sousal = sousal.loc[sousal['Année'].isin(["2012-2014", "2016-2018"])].loc[sousal['Zone'] == 'Thaïlande']
sousal = sousal.drop(['Code Domaine', 'Domaine', 'Code zone', 'Code Élément','Élément', 'Code Produit','Produit', 'Code année', 'Unité', 'Symbole', 'Description du Symbole', 'Note'],1)
sousal['Valeur'] = pd.to_numeric(sousal['Valeur'], errors='coerce')
sousal['Valeur'] *= 1000000
sousal['Année'] = [2013,2017]
sousal

Zone  Année     Valeur
1117  Thaïlande   2013  5600000.0
1121  Thaïlande   2017  5400000.0

In [93]:
sousal.iloc[0,2]/popu_thai.loc[1279,'Valeur']*100

8.218004282607481

<span style="color: blue">Soit 8,2% de sous-alimentés en 2013</span>

In [94]:
sousal.iloc[1,2]/popu_thai.loc[1283,'Valeur']*100

7.821834485839606

<span style="color: blue">Soit 7,8% de sous-alimentés en 2017</span>